### DISCUSSION TITLE
Allure ID is a disaster. Why it keeps changing? Why I cannot assign any number to it?

### DISCUSSION THREAD

**Discussion by PavloKokhaniuk** [allure-framework/allure2#1976] [TestOps]
Every time our test gets an update allure changes its ID. So we assigned Allure.id to every case, but no, we needed to assign specific number that allure already had in logs, ok, another reassign but noo you need to assign this specific ID that allure already assigned to this case, but will change this ID if there will be any change in case. Wouldn't even consider this system if I had know that it have issues with handling as simple feature as ID for cases that can change its name or folder.

**Comment by baev** [comment]
Please add additional details on the language and framework you are using. Some code examples would be the best

**Reply by uaxpasha** [reply]
Main problem - cannot reassign id.
Case: several suites runned simultaneously
so suite 1 recieve id in test like:1,3,7,
suite 2 received id 2,4,5,6 
(or another case: tests3 that should have ran after test2 received id 2. So I have test3 with id 2, and test2 with id 3, no idea why it's happened, but I want to fix them ids).
So would be great to have ability reassign id to whatever case I want, without restriction.

**Comment by cheshi-mantu** [comment]
@PavloKokhaniuk I think this might help: https://help.qameta.io/support/solutions/articles/101000480600-cooking-the-allureid

**Reply by ameduza** [reply]
@cheshi-mantu , @baev  Hi guys, thanks for the support page. It definitely reveals some details on how AllureID is treated inside the TestOps. Even though I'm still not so clear of the expected workflow from my side. I'm currently evaluating Allure TestOps for our team and struggling a bit with a clear picture of my possible workflow.My project contains Gherkin features (BDD) that are implemented with Cypress. From what I learned, in a fresh TestOps project I can create test cases out of the automation code execution (through Launches=>Upload new result OR via CI integration). At this point TestOps will assign allureID to each scenario. 
My question is - am I right that after initial import I have to go back to my features and set @testID() tag (allureID reference in cypress-allure-plugin) with generated values to fixate IDs for ALL scenarios? I mean is it really the only proper workflow?

Also, I have a side question on how to deal with parameterized tests aka ScenarioOutlines. From what I see every row (example) in Examples table under ScenarioOutline is considered a unique allureID test case. Can you please suggest how I can fixate allureIDs for ScenarioOutlines?

Thanks in advance,
Aleksandr

**Reply by baev** [reply]
Usually, you run your tests, and results will be mapped to test cases automatically by using test coordinates (for your setup, it should be something like feature file path + feature name + scenario name). If you want to change the coordinates (e.g. rename the scenario or move the feature file to some other path), you'll need to specify `allure.id` for each affected scenario. 

Managing ids manually can be annoying, so we provide an IDEA plugin that will set `allure.id` automatically to each scenario. But not yet all the frameworks are supported by the IDEA plugin at the moment, and I'm not sure about Gherkin's features; as far as I know, it's not yet supported. Although if you are interested, please feel free to submit the request to our support portal https://help.qameta.io 

Besides that, there is always an option to create a test case in Allure TestOps first and then link it by id with each scenario, the same way other TMS systems force you to work with automation.

**Reply by baev** [reply]
BTW `Cypress` is 3rd party integration, so that you may experience some issues with it. If it's working not the way I have described above, please let us know what Cypress integration (package coordinates + version) you're using, and we will look at it.

**Reply by baev** [reply]
> Also, I have a side question on how to deal with parameterized tests aka ScenarioOutlines. From what I see every row (example) in Examples table under ScenarioOutline is considered a unique allureID test case. Can you please suggest how I can fixate allureIDs for ScenarioOutlines?

It may be the issue in Cypress integration, see the comment above. By default it should work the same way for all scenario types

**Reply by ameduza** [reply]
Hi @baev Dmitry, thanks for the quick response.

> ... so we provide an IDEA plugin that will set allure.id automatically to each scenario

IDEA is a tool not everyone work with, I'm afraid. Intellij IDEA is Java one, for Cypress development (Typescript) we use VSCode at the team. Just curious, do you have this plugin for WebStorm? 
Seems that plugin is not smth I can consider as an option for now. :(

> Besides that, there is always an option to create a test case in Allure TestOps first and then link it by id with each scenario, the same way other TMS systems force you to work with automation.

Import actually is a separate topic. I tried Test Cases => Import test cases, but no luck. Do you have an article explaining the CSV expected format and the entire procedure?

**Reply by baev** [reply]
> Import actually is a separate topic. I tried Test Cases => Import test cases, but no luck. Do you have an article explaining the CSV expected format and the entire procedure?

Feel free to create a separate discussion for that!

**Reply by baev** [reply]
JFYI We plan to support all Intellij IDEs as well as VSCode.
